src/gui/search/searchwidget.cpp

```diff
@@ -242,12 +241,18 @@ QTabWidget *SearchWidget::searchTabs() const
 void SearchWidget::on_searchButton_clicked()
 {
     if (Utils::Misc::pythonVersion() < 0) {
-        m_mainWindow->showNotificationBaloon(tr("Search Engine"), tr("Please install Python to use the Search Engine."));
+        // unconditional notifications, user can not disable it
+        Notifications::Request()
+            .title(tr("Search Engine"))
+            .message(tr("Please install Python to use the Search Engine."))
+//          .setWidget(this)
+            .severity(Notifications::Severity::Error)
+            .exec();
```

Comment:
Finally I checked the entire PR. And I found the cause of my discontent. Here is the place!
But I still thought, why are you doing these cross-dependencies between the Manager and Request? Why is this method needed? (I mean `Request::exec()`)
Here you made a mistake of not implementing all this using the "Observer" pattern (as we decided before).
But why? Probably for the same reason you did another degradation I've noticed (I mean, SearchEngine in MainWindow).
This leads me to believe that we have to solve such problems in a different way. And in this case it's easier than you may think about.
MainWindow owns the SearchWidget and this is normal relationship. To follow Observer pattern NotificationsManager should recognize when the SearchWidget is created and connect to its signals (and to SearchEngine signals too; SearchWidget can own SearchEngine as it was before and it can have `SearchWidget::searchEngine()` property). How to do this? It's very simple. You just have to emit `MainWindow::searchWidgetCreated()` signal.